lib/git-process.ts

```diff
@@ -144,97 +145,133 @@ export class GitProcess {
-              code = GitNotFoundErrorCode
+              reject(err)
             }
 
-            const error = new Error(message) as ErrorWithCode
-            error.name = err.name
-            error.code = code
-            reject(error)
+            return
+          }
+
+          if (typeof code === 'number') {
+            resolve({ stdout, stderr, exitCode: code })
+            return
+          }
+
+          // Git has returned an output that couldn't fit in the specified buffer
+          // as we don't know how many bytes it requires, rethrow the error with
+          // details about what it was previously set to...
+          if (err.message === 'stdout maxBuffer exceeded') {
+            reject(
+              new Error(
+                `The output from the command could not fit into the allocated stdout buffer. Set options.maxBuffer to a larger value than ${
+                  execOptions.maxBuffer
+                } bytes`
+              )
+            )
           } else {
             reject(err)
           }
+        })
 
-          return
-        }
+        pidResolve(spawnedProcess.pid)
 
-        if (typeof code === 'number') {
-          resolve({ stdout, stderr, exitCode: code })
-          return
-        }
+        ignoreClosedInputStream(spawnedProcess)
 
-        // Git has returned an output that couldn't fit in the specified buffer
-        // as we don't know how many bytes it requires, rethrow the error with
-        // details about what it was previously set to...
-        if (err.message === 'stdout maxBuffer exceeded') {
-          reject(
-            new Error(
-              `The output from the command could not fit into the allocated stdout buffer. Set options.maxBuffer to a larger value than ${
-                execOptions.maxBuffer
-              } bytes`
-            )
-          )
-        } else {
-          reject(err)
+        if (options && options.stdin !== undefined) {
+          // See https://github.com/nodejs/node/blob/7b5ffa46fe4d2868c1662694da06eb55ec744bde/test/parallel/test-stdin-pipe-large.js
+          spawnedProcess.stdin.end(options.stdin, options.stdinEncoding)
         }
-      })
-
-      ignoreClosedInputStream(spawnedProcess)
 
-      if (options && options.stdin !== undefined) {
-        // See https://github.com/nodejs/node/blob/7b5ffa46fe4d2868c1662694da06eb55ec744bde/test/parallel/test-stdin-pipe-large.js
-        spawnedProcess.stdin.end(options.stdin, options.stdinEncoding)
-      }
+        if (options && options.processCallback) {
+          options.processCallback(spawnedProcess)
+        }
+      }),
+      pidPromise
+    )
 
-      if (options && options.processCallback) {
-        options.processCallback(spawnedProcess)
-      }
-    })
+    return result as IGitTask
```

Comment:
You shouldn't need the `as IGitTask`, right? Since `result` is already an instance of `GitTask` which `implements IGitTask` 🤔 
```suggestion
    return result
```